Libraries/react-native/react-native.js

```diff
@@ -105,15 +119,60 @@ var ReactNative = {
 
   // See http://facebook.github.io/react/docs/addons.html
   addons: {
-    get LinkedStateMixin() { return require('LinkedStateMixin'); },
+    get LinkedStateMixin() {
+      if (__DEV__) {
+        addonWarn('LinkedStateMixin', 'react-addons-linked-state-mixin');
+      }
+      return require('LinkedStateMixin');
+    },
     Perf: undefined,
-    get PureRenderMixin() { return require('ReactComponentWithPureRenderMixin'); },
-    get TestModule() { return require('NativeModules').TestModule; },
+    get PureRenderMixin() {
+      if (__DEV__) {
+        addonWarn('PureRenderMixin', 'react-addons-pure-render-mixin');
+      }
+      return require('ReactComponentWithPureRenderMixin');
+    },
+    get TestModule() {
+      if (__DEV__) {
+        warning(
+          warningDedupe['TestModule'],
+          'React.addons.TestModule is deprecated. ' +
+          'Use ReactNative.NativeModules.TestModule instead.'
+        );
+        warningDedupe['TestModule'] = true;
+      }
+      return require('NativeModules').TestModule;
+    },
     TestUtils: undefined,
-    get batchedUpdates() { return require('ReactUpdates').batchedUpdates; },
-    get cloneWithProps() { return require('cloneWithProps'); },
-    get createFragment() { return require('ReactFragment').create; },
-    get update() { return require('update'); },
+    get batchedUpdates() {
+      if (__DEV__) {
+        warning(
+          warningDedupe['batchedUpdates'],
+          'React.addons.batchedUpdates is deprecated. ' +
+          'Use ReactNative.unstable_batchedUpdates instead.'
+        );
```

Comment:
This lint is stupid. If I didn't know I could, I shouldn't be working on this codebase. If I know when to make an exception I should be allowed to.